File: backend/app/api/v1/checkins.py
from typing import Annotated

from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.core.database import get_db
from app.models.user import User
from app.schemas.checkins import CheckInCreate, CheckInResponse
from app.services.checkins import create_checkin, get_latest_checkin, list_checkins

router = APIRouter(prefix="/checkins", tags=["Check-ins"])


@router.get("", response_model=list[CheckInResponse])
def read_checkins(
    current_user: Annotated[User, Depends(get_current_user)],
    db: Annotated[Session, Depends(get_db)],
):
    return list_checkins(current_user, db)


@router.post("", response_model=CheckInResponse, status_code=status.HTTP_201_CREATED)
def create_checkin_endpoint(
    payload: CheckInCreate,
    current_user: Annotated[User, Depends(get_current_user)],
    db: Annotated[Session, Depends(get_db)],
):
    return create_checkin(current_user, payload, db)


@router.get("/latest", response_model=CheckInResponse | None)
def read_latest_checkin(
    current_user: Annotated[User, Depends(get_current_user)],
    db: Annotated[Session, Depends(get_db)],
):
    return get_latest_checkin(current_user, db)

File: frontend/services/checkins.ts
import { api } from "@/lib/api";
import type { CheckIn, CheckInCreateRequest } from "@/types/checkins";

export async function getCheckIns(): Promise<CheckIn[]> {
  const response = await api.get<CheckIn[]>("/checkins");
  return response.data;
}

export async function getLatestCheckIn(): Promise<CheckIn | null> {
  const response = await api.get<CheckIn | null>("/checkins/latest");
  return response.data;
}

export async function createCheckIn(data: CheckInCreateRequest): Promise<CheckIn> {
  const response = await api.post<CheckIn>("/checkins", data);
  return response.data;
}
